src/main/java/teammates/ui/controller/InstructorCourseStudentDetailsEditSaveAction.java

```diff
@@ -63,7 +66,20 @@ public ActionResult execute() throws EntityDoesNotExistException {
                 logic.validateTeams(Arrays.asList(student), courseId);
             }
             
-            logic.updateStudent(studentEmail, student);
+            if (!isSectionChanged && isTeamChanged && isTeamChangedForWholeTeam) {
+                List<StudentAttributes> studentsInTeam =
+                        logic.getStudentsForTeam(originalStudentAttribute.getTeam(), courseId);
+                for (StudentAttributes studentInTeam : studentsInTeam) {
+                    if (studentInTeam.getEmail().equals(studentEmail)) {
+                        logic.updateStudentTeamChangedForWholeTeam(studentEmail, student);
+                    } else {
+                        studentInTeam.team = student.team;
+                        logic.updateStudentTeamChangedForWholeTeam(studentInTeam.getEmail(), studentInTeam);
```

Comment:
Also remember to check that the section limit for number of students isn't exceeded when the entire team is shifted to a different section